File: js/bank.mjs
import { Money } from "./money.mjs";
export class Bank {
  constructor() {
    this.exchangeRates = new Map();
  }
  addExchangeRate(currencyFrom, currencyTo, rate) {
    let key = currencyFrom + "->" + currencyTo;
    this.exchangeRates.set(key, rate);
  }
  convert(money, currency) {
    let currencyFrom = money.currency;
    let currencyTo = currency;
    if (currencyFrom === currencyTo) {
      return money;
    }
    let key = currencyFrom + "->" + currencyTo;
    let rate = this.exchangeRates.get(key);
    if (rate === undefined) {
      throw new Error(key);
    }
    let convertedAmount = new Money(money.amount * rate, currency);
    return convertedAmount;
  }
}

File: py/bank.py
from money import Money 
class Bank:

    def __init__(self):
        self.exchangeRates = {}

    def addExchangeRate(self, fromCurrency, toCurrency, rate):
        key = fromCurrency + "->" + toCurrency
        self.exchangeRates.update({key:rate})

    def convert(self, money, currency):
        if(money.currency == currency):
            return Money(money.amount, currency)
        key = money.currency + "->" + currency 
        if key in self.exchangeRates:
            return Money(money.amount * self.exchangeRates[key], currency)
        raise Exception(key)